javalib/src/main/scala/java/net/InetAddress.scala

```diff
@@ -0,0 +1,648 @@
+package java.net
+
+import scala.scalanative.native._
+import scala.scalanative.runtime.SocketHelpers
+
+import scala.collection.mutable.ArrayBuffer
+
+import java.util.StringTokenizer
+
+// Ported from Apache Harmony
+abstract class InetAddress private[net] (ipAddress: Array[Byte], host: String)
+    extends Serializable {
+  import InetAddress._
+
+  private[net] def this(ipAddress: Array[Byte]) = this(ipAddress, null)
+
+  def getHostAddress(): String = createIPStringFromByteArray(ipAddress)
+
+  def getHostName(): String = {
+    if (host == null) {
+      Zone { implicit z =>
+        val host_c = SocketHelpers.ipToHost(
+          toCString(createIPStringFromByteArray(ipAddress)),
+          true)
+        return fromCString(host_c)
+      }
+    }
+    return host
+  }
+
+  def getAddress() = ipAddress.clone
+
+  override def equals(obj: Any): Boolean = {
+    if (obj == null || obj.getClass != this.getClass)
+      return false
+
+    val objIPAddress = obj.asInstanceOf[InetAddress].getAddress;
```

Comment:
Actually this.getClass returns either Inet4Address or Inet6Address, so we can't avoid checking `this.getClass` here. We can if we override `equals` in both Inet4Address and Inet6Addres instead of just InetAddress as it is now. Do you think it's worth duplicating?